common/libs/contract.js

```diff
@@ -65,25 +65,16 @@ export default class Contract {
       throw new Error('Invalid number of arguments');
     }
 
-    return method.inputs
-      .map((input, idx) => this.encodeArg(input, args[idx]))
-      .join('');
+    const inputTypes = method.inputs.map(input => input.type);
+    return abi.rawEncode(inputTypes, args).toString('hex');
   }
 
-  encodeArg(input: ABITypedSlot, arg: any): string {
-    switch (input.type) {
-      case 'address':
-      case 'uint160':
-        assertString(arg);
-        return setLengthLeft(toBuffer(arg), 32).toString('hex');
-      case 'uint256':
-        if (arg instanceof Big) {
-          arg = '0x' + arg.toString(16);
-        }
-        assertString(arg);
-        return setLengthLeft(toBuffer(arg), 32).toString('hex');
-      default:
-        throw new Error(`Dont know how to handle abi type ${input.type}`);
-    }
+  decodeArgs(method: ABIMethod, argData: string) {
+    // Remove method selector from data, if present
+    argData = argData.replace(`0x${this.getMethodSelector(method)}`, '');
+    // Convert argdata to a hex buffer for ethereumjs-abi
+    const argBuffer = new Buffer(argData, 'hex');
+    // Decode!
+    return abi.rawDecode(this.getMethodTypes(method), argBuffer);
```

Comment:
The `erc20` tests will show that this works with this implementation. However, I should definitely add some tests for this function directly, not just the `erc20` one.

`ethereumjs-abi`'s docs are really bad because they show functions that _don't exist yet_. The only ones that exist for this are `rawEncode` and `rawDecode` which cannot handle full encodes, only params. The `encode/decode` and `simpleEncode/simpleDecode` functions they have in the docs are just proposals, not actually working.

If you have advice on a better way to use the library, or a different library that handles this more gracefully, I'm all ears.